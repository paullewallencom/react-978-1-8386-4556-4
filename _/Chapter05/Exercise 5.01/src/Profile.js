import React, { Component } from "react";
// styles
import "./profile.css";

class Profile extends Component {
  render() {
    const { name, image, age, interests, color, movie } = this.props.user;
    return (
      <section className="profile">
        <header>
          <h1>{name}</h1>
        </header>
        <div className="profile-content">
          <div className="profile-image">
            <img src={image} alt={name} />
          </div>
          <div>
            <p>
              <strong>Age:</strong> {age}
            </p>
            <p>
              <strong>Interests:</strong> {interests}
            </p>
            <p>
              <strong>Favourite Color:</strong> {color}
            </p>
            <p>
              <strong>Favourite Movie:</strong> {movie}
            </p>
          </div>
        </div>
      </section>
    );
  }
}
export default Profile;
