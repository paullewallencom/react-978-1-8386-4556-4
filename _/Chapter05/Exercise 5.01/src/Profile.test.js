import React from "react";
import { render, cleanup } from "@testing-library/react";
import Profile from "./Profile";

afterEach(cleanup);

const user = {
  name: "Brian",
  interests: "Reading, Swimming, Technology",
  age: 9,
  image:
    "https://images.unsplash.com/photo-1470071131384-001b85755536?auto=format&fit=crop&w=200&q=80",
  color: "Red",
  movie: "Star Wars"
};

const withMarkup = query => text =>
  query((content, node) => {
    const hasText = node => node.textContent === text;
    const childrenDontHaveText = Array.from(node.children).every(
      child => !hasText(child)
    );
    return hasText(node) && childrenDontHaveText;
  });

it("renders profile", () => {
  const { asFragment } = render(<Profile user={user} />);
  expect(asFragment()).toMatchSnapshot();
});

it("renders name", () => {
  const { getByText } = render(<Profile user={user} />);
  expect(!!withMarkup(getByText)(user.name)).toBe(true);
});

it("renders Age", () => {
  const { getByText } = render(<Profile user={user} />);
  expect(!!withMarkup(getByText)("Age: " + user.age)).toBe(true);
});

it("renders Interests", () => {
  const { getByText } = render(<Profile user={user} />);
  expect(!!withMarkup(getByText)("Interests: " + user.interests)).toBe(true);
});

it("renders Favourite Color", () => {
  const { getByText } = render(<Profile user={user} />);
  expect(!!withMarkup(getByText)("Favourite Color: " + user.color)).toBe(true);
});

it("renders Favourite Movie", () => {
  const { getByText } = render(<Profile user={user} />);
  expect(!!withMarkup(getByText)("Favourite Movie: " + user.movie)).toBe(true);
});
