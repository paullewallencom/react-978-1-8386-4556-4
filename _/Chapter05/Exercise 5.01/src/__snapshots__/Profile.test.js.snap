// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders profile 1`] = `
<DocumentFragment>
  <section
    class="profile"
  >
    <header>
      <h1>
        Brian
      </h1>
    </header>
    <div
      class="profile-content"
    >
      <div
        class="profile-image"
      >
        <img
          alt="Brian"
          src="https://images.unsplash.com/photo-1470071131384-001b85755536?auto=format&fit=crop&w=200&q=80"
        />
      </div>
      <div>
        <p>
          <strong>
            Age:
          </strong>
           9
        </p>
        <p>
          <strong>
            Interests:
          </strong>
           Reading, Swimming, Technology
        </p>
        <p>
          <strong>
            Favourite Color:
          </strong>
           Red
        </p>
        <p>
          <strong>
            Favourite Movie:
          </strong>
           Star Wars
        </p>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
